### DISCUSSION TITLE
Strange Behaviour Solo Gamecube Controller

### DISCUSSION THREAD

**Discussion by LarsWerkman** [darthcloud/BlueRetro#27] [Q&A] [answered]
### Background
I planning on developing a solo controller bluetooth adapter board for the GameCube. Currently I'm building the prototype hardware on a breadboard (that might be part of my problem)

### Hardware
Esp-Wroom-32-Dev-Kit 38pin (Not the 30 pin as in the schematic below)
<img width="1425" alt="Screenshot 2021-04-20 at 00 32 07" src="https://user-images.githubusercontent.com/2527778/115311447-f6dc3300-a16f-11eb-8e69-f289b52935e0.png">

### Firmware
BlueRetro Latest version (2021-04-19), using the `gc_spiffs` config file

### Issue
When connected over USB, pairing a controller works perfectly (debugging input logs works fine), When it is connected only using the Gamecube pairing the controller won't work (even if it has paired before)

### Suggestions / Questions
Would it be possible list/un-pair controllers with the web-tool?

Thanks for the help!

**Comment by darthcloud** [comment]
What voltage do you read on ESP32 VCC input ( 5V input LDO) with a multimeter?

Sometimes cable extension are poor quality make sure you read 5V from the cable alone.

Make sure VCC diode is not backwards.

Pressing boot btn (IO0) will disconnect all controller ( except Xbox that one require shutdown via xbox btn)

I could had a page for that someday when I rework the webcfg.

**Reply by darthcloud** [reply]
Not sure which dev board you got, any pictures?

**Reply by LarsWerkman** [reply]
I checked the voltages and indeed the problem seems to be there:
At the cable: `4.9V`
At the VIN of the board: `4.4V` <- voltage drop here is also pretty high maybe the random diode I grabbed isn't that efficient
At the 3.3V Out: `2.8V` <- I think this is the problem (as the system requires a proper 3.3V source)

Picture of the board
![IMG_20210419_212928](https://user-images.githubusercontent.com/2527778/115365335-94615200-a1c4-11eb-8d68-14ac030c0262.jpeg)

**Reply by darthcloud** [reply]
Remove the diode, should be safe without usb.

**Reply by LarsWerkman** [reply]
Removed the diode, That seems to have fixed the voltages which are now stable at `~4.9` and the 3v3 at precisely `3.3v`.
Altough didn't fix the issue I exprienced.. the bluetooth controller still didn't want to pair, and the error light keeps on pulsing in I would say ~500 milliseconds

(Would there be a way to debug this without connecting over usb?, maybe wifi)

**Comment by darthcloud** [comment, marked as answer]
If LED flash that means the board reboot in loop probably the floating P2-P4 pins generate too much interrupt.

Connect unused plug pin to 3.3v as described here: https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-Cables-Build-Instructions#assembly-instructions-25

You would need a external usb 3.3V uart cable and connect it directly to TXD/GND of your ESP32 board to get the log to bypass USB power.

**Reply by darthcloud** [reply]
Base on schematic that was probably already fine.
Try removing it from breadboard. If you use gc_spiffs you should be able to connect only 5v , io19 and GND and it should work

**Reply by darthcloud** [reply]
Gave gc_spiffs a try (0.15.1) with a esp DevKitC with only 3 wire, work fine for me.

So I think its likely breadboard issue.

![PXL_20210421_105605725 MP](https://user-images.githubusercontent.com/3744056/115543022-ebbbfc80-a26e-11eb-8686-eedc02743565.jpg)

**Reply by LarsWerkman** [reply]
Hmm interesting I will try that..

I did some more debuggin with a USB UART like you suggested and got a `Brownout detector was triggered` error, Which I find really strange as I checked with the multimeter that the voltage is 3.3V as it should be

**Reply by darthcloud** [reply]
ok definitely a power issue then even if voltage is good it might not have enough amp.

Using a regular GC wired controller check if the rumble work, rumble is power from 5V.

**Reply by LarsWerkman** [reply]
Wiring it up with only 3 wires seemed too also have solve the pairing problem for me!! thank you so much for the help.

right now i'm only facing some input problems with my 8Bitdo SN30 controller ~~(maybe I just need to map the controls)~~ My switch pro controller works great, so I will dive into the mapping

**Reply by darthcloud** [reply]
Associate it in xinput mode (X + start i think)

in dinput you will have weird mapping by default, and maybe it a bug if not using latest 0.15.1
